src/components/DropdownButton.vue

```diff
@@ -0,0 +1,160 @@
+<template>
+  <div class="relative block max-w-min">
+    <!-- rounded-X-none is required to fight Edge UA styles that apply a 2px border radius to all `button` elements -->
+    <div class="flex">
+      <slot
+        :button-props="{
+          class: 'dropdown-button rounded-l-sm rounded-r-none',
+          type: 'button',
+        }"
+      />
+      <button
+        ref="dropdownButton"
+        type="button"
+        class="dropdown-button ml-1 rounded-r-sm rounded-l-none"
+        :class="{ 'dropdown-button-active': isOpen }"
+        aria-haspopup="menu"
+        :aria-label="$t('dropdown-button.aria.arrow-label')"
+        :aria-expanded="isOpen"
+        @click="toggleOpen"
+        @keydown.space.prevent="toggleOpen"
+      >
+        <svg class="h-2 w-4">
+          <use :href="`${icons.caretDown}#icon`" />
+        </svg>
+      </button>
+    </div>
+
+    <div
+      ref="dropdownContainer"
+      class="dropdown-container"
+      :class="{ hidden: !isOpen }"
+      role="menu"
+      :aria-hidden="!isOpen"
+      @focusout="onFocusout"
+    >
+      <slot
+        name="items"
+        :item-class="'dropdown-item'"
+        :item-a11y-props="{ role: 'menuitem' }"
+        :toggle-open="toggleOpen"
+        :active-item-class="'dropdown-item-active'"
+        :on-item-keydown="onItemKeydown"
+      />
+    </div>
+  </div>
+</template>
+
+<script>
+import caretDown from '~/assets/icons/caret-down.svg'
+
+const DropdownButton = {
+  name: 'DropdownButton',
+  data: () => ({
+    isOpen: false,
+    icons: {
+      caretDown,
+    },
+  }),
+  mounted() {
+    document.addEventListener('click', this.onClickout)
+  },
+  beforeDestroy() {
+    document.removeEventListener('click', this.onClickout)
+  },
+  beforeUnmount() {
+    document.removeEventListener('click', this.onClickout)
+  },
+  methods: {
+    getItems() {
+      return Array.from(
+        this.$refs.dropdownContainer.querySelectorAll('[role="menuitem"]')
+      )
+    },
+    onClickout(e) {
```

Comment:
Should use the identifier `event` to be consistent with the `onItemKeydown` handler.
```suggestion
    onClickout(event) {
```